activerecord/lib/active_record/core.rb

```diff
@@ -314,7 +314,8 @@ def find_by(*args) # :nodoc:
         end
 
         keys = hash.keys
-        statement = cached_find_by_statement(keys) { |params|
+        cache_key = keys.then { |keys| keys.length == 1 ? keys.first : keys }
```

Comment:
```suggestion
        cache_key = keys.length == 1 ? keys.first : keys 
```
it looks same work. is `#then` unnecessary?

Reply:
Totally.

I was trying to remember if there was a reason, I believe it's because I initially had this line:

```ruby
keys = hash.keys.then { |keys| keys.length == 1 ? keys.first : keys }
```

Which without then, would have looked like

```ruby
keys = hash.keys.length == 1 ? hash.keys.length : hash.keys.first
```

And the first one looked a bit nicer to me, but that didn't matter because we need `keys` to be an array on like 319 anyway.

With that said, I think this is all about to go away anyway if we go with the other approach of making the change in `find`